src/mlpack/methods/bayesian_ridge/bayesian_ridge.hpp

```diff
@@ -0,0 +1,297 @@
+
+  /**
+   * Run BayesianRidge regression. The input matrix (like all mlpack matrices)
+   * should be
+   * column-major -- each column is an observation and each row is a dimension.
+   * 
+   * @param data Column-major input data
+   * @param responses A vector of targets.
+   * @return score. Root Mean Square Error. Equal to -1 of two feature vectors 
+   *    or more are colinear.
+   **/
+  double Train(const arma::mat& data,
+               const arma::rowvec& responses);
+
+  /**
+   * Predict \f$y_{i}\f$ for each data point in the given data matrix using the
+   * currently-trained Bayesian Ridge model.
+   *
+   * @param points The data points to apply the model.
+   * @param predictions y, Contain the  predicted values on completion.
+   **/
+  void Predict(const arma::mat& points,
+               arma::rowvec& predictions) const;
+
+  /**
+   * Predict \f$y_{i}\f$ and the standard deviation of the predictive posterior 
+   * distribution for each data point in the given data matrix, using the
+   * currently-trained Bayesian Ridge estimator.
+   *
+   * @param points The data point to apply the model.
+   * @param predictions y, which will contained calculated values on completion.
+   * @param std Standard deviations of the predictions.
+   */
+  void Predict(const arma::mat& points,
+               arma::rowvec& predictions,
+               arma::rowvec& std) const;
+
+   /**
+   * Compute the Root Mean Square Error
+   * between the predictions returned by the model
+   * and the true repsonses.
+   *
+   * @param Points Data points to predict
+   * @param responses A vector of targets.
+   * @return RMSE
+   **/
+  double Rmse(const arma::mat& data,
+              const arma::rowvec& responses) const;
+
+  /**
+   * Center and scaleData the data. The last four arguments
+   * allow future modifation of new points.
+   *
+   * @param data Design matrix in column-major format, dim(P,N).
+   * @param responses A vector of targets.
+   * @param centerData If true data will be centred according to the points.
+   * @param centerData If true data will be scales by the standard deviations
+   *     of the features computed according to the points.
+   * @param dataProc data processed, dim(N,P).
+   * @param responsesProc responses processed, dim(N).
+   * @param dataOffset Mean vector of the design matrix according to the 
+   *     points, dim(P).
+   * @param dataScale Vector containg the standard deviations of the features
+   *     dim(P).
+   * @param reponses_offset Mean of responses.
+   */
+  void CenterScaleData(const arma::mat& data,
+                       const arma::rowvec& responses,
+                       const bool centerData,
+                       const bool scaleData,
+                       arma::mat& dataProc,
+                       arma::rowvec& responsesProc,
+                       arma::colvec& dataOffset,
+                       arma::colvec& dataScale,
+                       double& responsesOffset);
```

Comment:
Maybe it's more clean to return the `responsesOffset` instead of passing it as an argument.

Reply:
Done